### PR TITLE
added tests for most functions

### PR DESCRIPTION
**Pull request by seanheinrichs**
image

Still missing test for Ali. How exactly are we supposed to call his function? It's not taking a list.

**All tests currently pass**, this is just to show "UI" for a failure.

btw I had to round some of the results because the floating point arithmetic was causing overflow. tried to keep as much precision as possible

### CODE REVIEW

File: run_tests.py [hunk 1]
```diff
@@ -1,74 +1,208 @@
...
                     return False
     return True
 
+# mean absolute deviation test for integer cases
+def mad_test_integer():
+    print(f'{bcolors.HEADER}mad_integer test starting...')
+    for i in range(0, 10): 
+        list = []
+        for j in range(0, 10):
+            x = random.randint(0, 100)
```
Comment (steveiaco) [inline run_tests.py:115]: I've never seen unit tests use random inputs, but then again, I haven't seem many unit tests. 

Do you know if it's a good idea to do this? It'd make things a little unreproducable if we encountered a strange edge case?

**Review by steveiaco — APPROVED** [review]
Everything looks good! There's a function and it's associated tests missing, do you know which that is?

File: run_tests.py (hunk 1 above)
Comment (seanheinrichs) [reply run_tests.py:115]: Honestly, I think we should redo all tests and instead of doing random numbers, focus on 1-2 normal tests and then only do edge cases. Having ten of these tests is useless, but good enough for iteration 1 IMO.
